### PR TITLE
Restructuring proposal

### PR DESCRIPTION
**Pull request by mBialczak**
Some proposal of new structure.
Maybe not perfect yet.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ COPY . /application
 
 RUN cmake -S . -B build && cmake --build build
 
-RUN /application/bin/${PROJECT_NAME}-ut
+RUN /application/bin/${PROJECT_NAME}-all-tests
```
Comment (fbtom) [inline Dockerfile:14]: simply `tests`

File: src/tests/ut/receptionist/shift_test.cpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #include "shift_test.hpp"
+#include "shift.hpp"
```
Comment (fbtom) [inline src/tests/ut/receptionist/shift_test.cpp:2]: It's included in header, not needed here

File: src/tests/ut/receptionist/shift_test.cpp [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
 
-namespace
+namespace unit_test::receptionist
```
Comment (fbtom) [inline src/tests/ut/receptionist/shift_test.cpp:4]: Please revert it to anonymous namespace 

[cpp coreguidlines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf22-use-an-unnamed-anonymous-namespace-for-all-internalnon-exported-entities)

If we want to include namespaces we shall speak about it's structure and start from source code

File: src/tests/ut/visit/visit_test.cpp [hunk 4]
```diff
@@ -1,7 +1,10 @@
-#include "../app/doctor/doctor.hpp"
-#include "../app/visit/visit.hpp"
+#include "doctor.hpp"
+#include "visit.hpp"
 #include "gtest/gtest.h"
 
+namespace unit_test::visit
```
Comment (fbtom) [inline src/tests/ut/visit/visit_test.cpp:5]: This isn't a scope of this change, but could you create a header file  and move structure over there and keep tests here with anonymous namespace?

File: src/tests/ut/receptionist/shift_test.hpp [hunk 5]
```diff
@@ -1,7 +1,11 @@
 #pragma once
-#include "../../../app/receptionist/shift.hpp"
+
+#include "shift.hpp"
 #include "gtest/gtest.h"
 
+namespace unit_test::receptionist
```
Comment (fbtom) [inline src/tests/ut/receptionist/shift_test.hpp:6]: remove in hpp, only tests should be in anonymous namespace

**Review by fbtom — CHANGES_REQUESTED** [review]

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

File: src/tests/ut/receptionist/shift_test.cpp (hunk 3 above)
Comment (mBialczak) [reply src/tests/ut/receptionist/shift_test.cpp:4]: Done

File: src/tests/ut/receptionist/shift_test.hpp (hunk 5 above)
Comment (mBialczak) [reply src/tests/ut/receptionist/shift_test.hpp:6]: done

File: src/tests/ut/visit/visit_test.cpp (hunk 4 above)
Comment (mBialczak) [reply src/tests/ut/visit/visit_test.cpp:5]: done

File: Dockerfile (hunk 1 above)
Comment (mBialczak) [reply Dockerfile:14]: done

**Comment by ziobron** [comment]
Thanks @fbtom for Code Review.
🏅 1 XP granted

**Review by BaqWin — APPROVED** [review]

**Comment by ziobron** [comment]
Thanks @BaqWin for Code Review.
🏅 1 XP granted

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @mBialczak for making course materials better!

**Pull request merged**